### ISSUE TITLE
The convert_folder2imcfolder function returns AssertionError: assert(len(mcd_files) == 1)

### ISSUE THREAD

**Issue by simonewebb** [BodenmillerGroup/imctools#60]
Hello

Firstly, thanks so much for producing such a well documented and useful tool! We appreciate your work.

I have a question regarding the imcpreprocessing.iynb tutorial. I am currently trying to convert 3 .mcd file into .tiffs for a project I am working on. I have saved the .mcd files in the following format (trying to stay as close as possible to the format you gave for the 2 fluidigm example datasets):

..../bm/figs/hyperion_mcd_files/F67Bone.zip/F67Bone.mcd
..../bm/figs/hyperion_mcd_files/F69Bone.zip/F69Bone.mcd
..../bm/figs/hyperion_mcd_files/F70Bone.zip/F70Bone.mcd

When I run the "convert mcd containing folders into imc zip folders" cell in the tutorial I am met with an assertion error: AssertionError: assert(len(mcd_files) == 1). Details in screenshot.

Could this be because I am not able to provide the "associated .txt file generated during the acquisition of this .mcd file". Or is this due to another issue? I am able to run the tutorial notebook successfully with the test dataset

Thanks in advance for your help!

Simone
![Screenshot 2019-10-25 at 16 07 07](https://user-images.githubusercontent.com/44371055/67583065-c108ea80-f742-11e9-87d6-0fd27eeb3fb0.png)

**Comment by votti** [comment]
Hi Simone,
The issue is that you have multiple `mcd` files per zip. The convention is to have each `mcd` (*optionally* together with all `.txt` files of that `.mcd`) in a *separate* `.zip` folder.

Then it should work.
Cheers, Vito

**Comment by simonewebb** [comment]
Hi Vito

Thanks so much for your quick response!

I have attached a screenshot of my directory structure. Within "hyperion_mcd_files" folder, i have 3 zip folders, which each only have one mcd within them (is this what are you referring to wrt "multiple mcd files per zip". If so I think I only have one per zip). In the screenshot you can see an example of the contents of the F70Bone.zip.

Thanks

Simone

**Comment by simonewebb** [comment]
> Hi Vito
> 
> Thanks so much for your quick response!
> 
> I have attached a screenshot of my directory structure. Within "hyperion_mcd_files" folder, i have 3 zip folders, which each only have one mcd within them (is this what are you referring to wrt "multiple mcd files per zip". If so I think I only have one per zip). In the screenshot you can see an example of the contents of the F70Bone.zip.
> 
> Thanks
> 
> Simone

<img width="901" alt="Screenshot 2019-10-25 at 16 26 07" src="https://user-images.githubusercontent.com/44371055/67584003-a5064880-f744-11e9-8657-81337e7e4cdf.png">

**Comment by votti** [comment]
Sorry that I did not read your report correctly!
You surely did follow things precisely.

Seing on the screenshot that you are using Mac I think I know what's going on:
Mac has tends to generate hidden (sub)folders and files in folders and zip folders that will also end in '.mcd' and thus are confusing the script.
This really reminds/motivates me that I need to fix this issue, so thanks for reporting: https://github.com/BodenmillerGroup/imctools/issues/43

Until I have fixed it, I would suggest you to:
1) open a terminal
2) Check the file content of the zip by typing: 'zipinfo EXACTPATHTO/bm/figs/hyperion_mcd_files/F67Bone.zip' -> I am pretty sure there should be also some hidden files in there
3) zip the files again from the unzipped files without any hidden files:
=> Using a command as described in the link above
 'zip -r mcd1.zip PATHTOMCD1/mcd1.mcd  -x ".*" -x "__MACOSX" '

Hope this helps.
Cheers!
